theopensystemslab/planx-new
# feat: setup Satellite basemap selection in MapAndLabel editor

**Key changes:**
- Adds basemap selection to editor modal
- Adjusts props on public component
- Sets up Mapbox access token env vars throughout
  - [x] `editor.planx.uk/.env` (key 1 is exclusively scoped to localhost:3000 & localhost:5173)
    - **Please note this is no longer committed!!** 
    - I've introduced an `editor.planx.uk/.env.test` and updated `.gitignore` & push/pull `pizza-secret` scripts
    - You may need to run `git rm -r --cached editor.planx.uk/.env` locally if VSCode isn't picking this change up
  - [x] Pulumi staging (key 2 is scoped to our public-facing domains only)
  - [x] Pulumi production (key 2)
  - [x] Github Action secrets (key 2)
![Screenshot from 2024-08-28 13-38-22](https://github.com/user-attachments/assets/46e6e3fe-5465-441d-ae50-d7af0fef49d8)

**Testing:**
- Go to the pizza editor & toggle on the `"TREES"` feature flag
- Create a Map & Label component with "Mapbox Satellite imagery" basemap selected
- Confirm the correct basemap and attribution displays
